src/storage/mod.rs

```diff
@@ -211,20 +190,23 @@ impl Storage {
     }
 
     pub fn stop(&mut self) -> Result<()> {
-        if self.thread.is_none() {
-            return Ok(());
-        }
-        try!(self.tx.send(Message::Close));
-        if self.thread.take().unwrap().join().is_err() {
-            return Err(box_err!("failed to wait storage thread quit"));
-        }
+        self.sched.take();
```

Comment:
@BusyJay The threadPool does not supply a `shutdown` method. Instead all threads will quit after the pool is dropped. Refer https://frewsxcv.github.io/rust-threadpool/src/threadpool/lib.rs.html#367